### PR TITLE
feat(log\trace):team、 log and trace init 

### PR DESCRIPTION
**Pull request by Qinyouzeng**
- team over-view
- log search
- trace list and detail

### CODE REVIEW

File: src/Services/Masa.Tsc.Service.Admin/Extensions/IElasticClientExtenstion.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    public static Dictionary<string, string>? AggResult(ISearchResponse<object> response, IEnumerable<RequestFieldAggregationDto> FieldMaps)
+    {
+        if (response.Aggregations == null || !response.Aggregations.Any())
+            return null;
+
+        var result = new Dictionary<string, string>();
+        foreach (var item in response.Aggregations)
+        {
+            var find = FieldMaps.First(m => m.Alias == item.Key);
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Extensions/IElasticClientExtenstion.cs:64]: find?

File: src/Services/Masa.Tsc.Service.Admin/Extensions/IElasticClientExtenstion.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            if (find.AggegationType - AggregationTypes.DistinctCount <= 0 && item.Value is ValueAggregate value && value != null)
+            {
+                string tem = default!;
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Extensions/IElasticClientExtenstion.cs:67]: tem?

File: src/Services/Masa.Tsc.Service.Admin/Infrastructure/EntityConfigurations/InstrumentEntityConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class InstrumentEntityConfiguration
+{
+    public void Configure(EntityTypeBuilder<Instrument> builder)
+    {
+        builder.HasKey(x => x.Id);
+        builder.Property(x => x.Sort).HasDefaultValue(0);
+        builder.Property(x => x.Name).HasMaxLength(200);
+        builder.Property(x => x.Layer).HasMaxLength(40);
+        builder.Property(x => x.Entity).HasMaxLength(40);
+        //builder.HasMany(x => x.Pannels).WithOne();
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Infrastructure/EntityConfigurations/InstrumentEntityConfiguration.cs:15]: Why comment it? If don't use it anymore, delete it.

File: src/Services/Masa.Tsc.Service.Admin/Infrastructure/EntityConfigurations/MetricEntityConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Tsc.Service.Admin.Infrastructure.EntityConfigurations;
+
+public class MetricEntityConfiguration
+{
+    public void Configure(EntityTypeBuilder<Domain.Aggregates.Metric> builder)
+    {
+        builder.HasKey(x => x.Id);
+        //builder.Property(x => x.Title).HasMaxLength(200).IsRequired();
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Infrastructure/EntityConfigurations/MetricEntityConfiguration.cs:11]: ?

File: src/Services/Masa.Tsc.Service.Admin/Infrastructure/Middleware/LogMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+//namespace Masa.Tsc.Service.Infrastructure.Middleware
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Infrastructure/Middleware/LogMiddleware.cs:4]: Why comment it?

File: src/Services/Masa.Tsc.Service.Admin/Infrastructure/Repositories/PannelRepository.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Tsc.Service.Admin.Infrastructure.Repositories;
+
+public class PannelRepository : Repository<TscDbContext, Pannel, Guid>, IPannelRepository
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Infrastructure/Repositories/PannelRepository.cs:6]: Panel

File: src/Services/Masa.Tsc.Service.Admin/Program.cs [hunk 7]
```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+var builder = WebApplication.CreateBuilder(args);
+
+var elasearchUrls = builder.Configuration.GetSection("Masa:Elastic:nodes").Get<string[]>();
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Program.cs:6]: elasearch?

File: src/Services/Masa.Tsc.Service.Admin/Program.cs [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+builder.Services.AddAuthClient(builder.Configuration["Masa:AuthUrl"]);
+builder.Services.AddPmClient(builder.Configuration["Masa:PmUrl"]);
+
+var app = builder.Services
+    // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+    .AddEndpointsApiExplorer()
+    .AddSwaggerGen(options =>
+    {
+        //var xmlFilename = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Program.cs:69]: ?

File: src/Services/Masa.Tsc.Service.Admin/Program.cs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                .UseEventBus()
+                .UseRepository<TscDbContext>();
+     })
+    .AddServices(builder);
+
+
+app.MigrateDbContext<TscDbContext>((context, services) =>
+{
+    //var logger = services.GetRequiredService<ILogger<TscDbContext>>();
+    //new AuthDbContextSeed().SeedAsync(context, logger).Wait();
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Program.cs:109]: ?

File: src/Services/Masa.Tsc.Service.Admin/Services/ProjectService.cs [hunk 10]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Tsc.Service.Admin.Services;
+
+public class ProjectService : ServiceBase
+{
+    public ProjectService(IServiceCollection services) : base(services, "/api/project")
+    {
+        App.MapGet($"{BaseUri}/over-view", OverViewAsync);
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Services/ProjectService.cs:10]: overview

File: src/Services/Masa.Tsc.Service.Admin/Services/TeamService.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+public class TeamService : ServiceBase
+{
+    public TeamService(IServiceCollection services) : base(services, "/api/team")
+    {
+        App.MapGet($"{BaseUri}/{{teamId}}/{{appid}}", GetAsync);
+    }
+
+    private async Task<TeamDto> GetAsync([FromServices] IEventBus eventBus, Guid teamId, string appId)
+    {
+        var teamQuery = new TeamDetailQuery(Guid.Empty, appId);
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Services/TeamService.cs:15]: The `teamId` has not been used?

File: src/Services/Masa.Tsc.Service.Admin/Services/TraceService.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        var query = new TraceDetailQuery(traceId);
+        await eventBus.PublishAsync(query);
+        return query.Result;
+    }
+
+    private async Task<PaginationDto<object>> GetListAsync([FromServices] IEventBus eventBus, RequestTraceListDto model)
+    {
+
+        ////var text = "{\"accounts\":[{\"accountID\":\"00000000-0000-0000-0000-000000000000\",\"accountNum\":\"test00001\",\"blocked\":0,\"city\":\"温州市\",\"countryRegionId\":\"CHN\",\"county\":\"乐清市\",\"creditMax\":0,\"creditRating\":\"\",\"currency\":\"CNY\",\"custClassificationId\":\"test一级代理商\",\"custGroup\":\"test国内非关联方\",\"custName\":\"test测试公司\",\"custStatus\":2,\"dlvMode\":\"\",\"dlvTerm\":\"\",\"inclTax\":1,\"invoiceAccount\":\"\",\"paymMode\":\"test支付宝\",\"paymTermId\":\"test月结\",\"priceGroup\":\"test经销公司\",\"state\":\"浙江省\",\"street\":\"北白象镇大桥工业园区\",\"taxGroup\":\"13%\",\"telephone\":\"18658198061\",\"custNumber\":\"18658198061\",\"ecsource\":0,\"custNumberNew\":\"\",\"refRecId\":0,\"salesInvoiceType\":0}]}";
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Services/TraceService.cs:29]: ?

File: src/Web/Masa.Tsc.Admin.Server/Program.cs [hunk 13]
```diff
@@ -0,0 +1,67 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+using Masa.BuildingBlocks.Identity.IdentityModel;
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin.Server/Program.cs:4]: Move these to _Imports.

File: src/Web/Masa.Tsc.Admin.Server/Program.cs [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+builder.Services.AddMasaIdentityModel(IdentityType.MultiEnvironment, options =>
+{
+    options.Environment = "environment";
+    options.UserName = "name";
+    options.UserId = "sub";
+})
+.AddMasaStackComponentsForServer("wwwroot/i18n", builder.Configuration["AuthServiceBaseAddress"], builder.Configuration["McServiceBaseAddress"])
+.AddMasaOpenIdConnect(builder.Configuration);
+
+
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin.Server/Program.cs:40]: Remove the blank line.

File: src/Web/Masa.Tsc.Admin.Server/TokenProvider.cs [hunk 15]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+//using Masa.Tsc.Contracts.Admin;
+
+//namespace Masa.Tsc.Admin.Server;
+
+//public class TokenProvider : ITokenProvider
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin.Server/TokenProvider.cs:8]: ?

File: src/Web/Masa.Tsc.Admin.Server/appsettings.Development.json [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Error",
+      "Microsoft.Hosting.Lifetime": "Error"
+    }
+  },
+
+  "OtlpUri": "http://10.10.90.114:32147",
+  "Masa": {
+    "Tsc": {
+      "ServiceName": "tsc.web",
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin.Server/appsettings.Development.json:19]: ?

File: src/Web/Masa.Tsc.Admin.WebAssembly/Program.cs [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+using Masa.Tsc.Admin.Rcl;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+
+var builder = WebAssemblyHostBuilder.CreateDefault(args);
+builder.RootComponents.Add<App>("#app");
+builder.RootComponents.Add<HeadOutlet>("head::after");
+builder.Services.AddMasaBlazor();
+builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+//builder.Services.AddSingleton<WeatherForecastService>();
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin.WebAssembly/Program.cs:13]: ?

File: src/Web/Masa.Tsc.Admin/Data/Trace/TraceDetailModel.cs [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+        if (value == null)
+        {
+            Reset();
+            return;
+        }
+        Current = value;
+
+        var name = Current.ContainsKey("transaction") ? "transaction" : "span";
+        Attributes = (Dictionary<string, object>)TscComponentBase.GetDictionaryValue(value, name);
+
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin/Data/Trace/TraceDetailModel.cs:43]: Remove the blank line

File: src/Web/Masa.Tsc.Admin/Extensitions/ObservableExtensions.cs [hunk 19]
```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Tsc.Admin.Rcl;
+
+public static class ObservableExtensions
+{
+    public static void AddObservable(this WebApplicationBuilder builder)
+    {
+
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin/Extensitions/ObservableExtensions.cs:10]: Remove the blank line.

File: src/Web/Masa.Tsc.Admin/Extensitions/ObservableExtensions.cs [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+            builder.IncludeFormattedMessage = true;
+            builder.ParseStateValues = true;
+            builder.AddOtlpExporter(option =>
+            {
+                option.Endpoint = url;
+            });
+        });
+
+        builder.Services.AddTscApiCaller(builder.Configuration["masa:tsc:apiUrl"]);
+
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin/Extensitions/ObservableExtensions.cs:54]: Remove the blank line.

File: src/Web/Masa.Tsc.Admin/Global/MenuConfigs.cs [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+internal class MenuConfigs
+{
+    private static Dictionary<string, Tuple<string, string>> _menus = new();
+
+    public static void Register()
+    {
+        //_menus.Add(new)
+    }
+
+    //public static void Register(string text, string url, string parent)
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin/Global/MenuConfigs.cs:15]: ?

File: src/Web/Masa.Tsc.Admin/Global/MenuConfigs.cs [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Masa.Tsc.Admin.Rcl;
+
+internal class MenuConfigs
+{
+    private static Dictionary<string, Tuple<string, string>> _menus = new();
+
+    public static void Register()
+    {
+        //_menus.Add(new)
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Admin/Global/MenuConfigs.cs:12]: ?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/ApiGateways/Caller/Masa.Tsc.Caller/Services/DirectoryService.cs [hunk 23]
```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Tsc.Caller.Services;
+
+public class DirectoryService : BaseService
+{
+    public DirectoryService(ICallerProvider caller) : base(caller, "/api/Instrument/directory") { }
```
[the remark this answers is not in the document]
Comment (Qinyouzeng) [reply src/ApiGateways/Caller/Masa.Tsc.Caller/Services/DirectoryService.cs]: Directory is a subfunction.

File: src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs [hunk 24]
```diff
@@ -1,13 +1,16 @@
...
 public class ChartViewDto
 {
     public string Title { get; set; }
 
     public string ChartType { get; set; }
-
-    public string[] Pointes { get; set; }
+    
+    /// <summary>
+    /// x value raw toString, like '8' 、'8.00'、'8.00%'
```
Comment (doddgu) [inline src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs:13]: x value?

File: src/Services/Masa.Tsc.Service.Admin/Application/Traces/QueryHandler.cs [hunk 25]
```diff
@@ -22,66 +22,51 @@ public QueryHandler(IServiceProvider provider, IElasticClient elasticClient, ILo
...
-                return sort.Ascending(ElasticConst.TraceTimestamp);
-            },
-            pageFn: () => Tuple.Create(false, 0, ElasticConst.MAX_DATA_COUNT - 1),
-            resultFn: (rep, q) =>
-            {
-                q.Result = rep.Documents!;
-            },
+            condition: (container, q) => container.Term(t => t.Field(ElasticConst.TRACE_ID).Value(q.TraceId)),
+            sort: (sort, q) => sort.Ascending(ElasticConst.TraceTimestamp),
+            pageration: () => ValueTuple.Create(false, 0, ElasticConst.MAX_DATA_COUNT - 1),
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Application/Traces/QueryHandler.cs:27]: pageration?

File: src/Services/Masa.Tsc.Service.Admin/Program.cs [hunk 26]
```diff
@@ -92,6 +90,7 @@
             }
         });
     })
+    .AddTransient(typeof(IMiddleware<>), typeof(LogMiddleware<>))
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Program.cs:93]: code format

File: src/Services/Masa.Tsc.Service.Admin/Services/TraceService.cs [hunk 27]
```diff
@@ -50,7 +37,7 @@ private async Task<IEnumerable<string>> GetAttrValuesAsync([FromServices] IEvent
         return query.Result;
     }
 
-    private async Task<ChartLineDataDto<ChartPointDto>> GetAggegationAsync([FromServices] IEventBus eventBus, [FromBody] RequestAggregationDto param)
+    private async Task<ChartLineDataDto<ChartPointDto>> GetAggregateAsync([FromServices] IEventBus eventBus, [FromBody] RequestAggregationDto param)
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/Services/TraceService.cs:40]: Get is v, Aggregate is v too.

File: src/Services/Masa.Tsc.Service.Admin/appsettings.Develop.json [hunk 28]
```diff
@@ -7,7 +7,39 @@
       "Microsoft.Hosting.Lifetime": "Error"
     }
   },
-  "AllowedHosts": "*",
-  "AuthServiceBaseAddress": "https://auth-service-develop.masastack.com/"
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=sqlserver-dev-svc.stack;Database=masa-tsc;User Id=sa;Password=p@ssw0rd;"
+  },
+
+  "Masa": {
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/appsettings.Develop.json:14]: Masa is same level with DccOptions?

File: src/Services/Masa.Tsc.Service.Admin/appsettings.Development.json [hunk 29]
```diff
@@ -13,13 +13,13 @@
...
     "Tsc": {
-      "ServiceName": "masa.tsc.api",
+      "ServiceName": "masa-tsc-service",
       "ServiceNameSpace": "Development",
       "ServiceVersion": "0.1.0"
     },
     "AuthUrl": "https://auth-service-develop.masastack.com",
     "PmUrl": "https://pm-service-develop.masastack.com",
-    "OtlpUrl": "http://10.10.90.114:32147",
+    "OtlpUrl": "http://otel.lonsid.cn:32247",
```
Comment (doddgu) [inline src/Services/Masa.Tsc.Service.Admin/appsettings.Development.json:22]: Is this internal domain?

File: src/Web/Masa.Tsc.Web.Admin.Server/Pages/Error.cshtml.cs [hunk 30]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Server.Pages;
+
+[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin.Server/Pages/Error.cshtml.cs:6]: Why need ResponseCache?

File: src/Web/Masa.Tsc.Web.Admin.Server/appsettings.Development.json [hunk 31]
```diff
@@ -12,11 +12,10 @@
       "Microsoft.Hosting.Lifetime": "Error"
     }
   },
-
-  "OtlpUri": "http://10.10.90.114:32147",
+  "otlpUrl": "http://otel.lonsid.cn:32247",
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin.Server/appsettings.Development.json:15]: Is this internal domain?

File: src/Web/Masa.Tsc.Web.Admin/Shared/MainLayout.razor [hunk 32]
```diff
@@ -0,0 +1,88 @@
...
+    <MErrorHandler ShowDetail=false>
+    @Body
+    </MErrorHandler>
+    </Layout>*@
+
+@code
+{
+    [Inject]
+    private BlazorComponent.I18n.I18n I18n { get; set; } = default!;
+    }
```
Comment (doddgu) [inline src/Web/Masa.Tsc.Web.Admin/Shared/MainLayout.razor:88]: code format

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Web/Masa.Tsc.Web.Admin.Server/appsettings.Development.json (hunk 31 above)
Comment (Qinyouzeng) [reply src/Web/Masa.Tsc.Web.Admin.Server/appsettings.Development.json:15]: No, 域名的现在外网可访问，ip是内网，我已经告诉了王静克，他明天再调整下

Comment (Qinyouzeng) [reply src/Web/Masa.Tsc.Web.Admin.Server/appsettings.Development.json:15]: 确认了，只有内网可以访问

File: src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs (hunk 24 above)
Comment (Qinyouzeng) [reply src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs:13]: echart数据展示时，x轴数据和y轴数据需要分开设置，这个地方设置的为y轴的纵坐标数据

File: src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs [hunk 33]
```diff
@@ -0,0 +1,16 @@
...
+namespace Masa.Tsc.Contracts.Admin.Charts;
+
+public class ChartViewDto
+{
+    public string Title { get; set; }
+
+    public string ChartType { get; set; }
+    
+    /// <summary>
+    /// y value raw toString, like '8' 、'8.00'、'8.00%'
```
Comment (doddgu) [inline src/Contracts/Masa.Tsc.Contracts.Admin/Charts/ChartViewDto.cs:13]: What's the mean of Y?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**